src/unistd.rs

```diff
@@ -1754,6 +1754,40 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
     Ok((fd, PathBuf::from(pathname)))
 }
 
+/// Creates a directory which persists even after process termination
+///
+/// * `template`: a path whose 6 rightmost characters must be X, e.g. `/tmp/tmpdir_XXXXXX`
```

Comment:
Why 6?  mkdtemp(3) imposes no such limitation.

Reply:
It says in the manual "at least 6" like mkstemp

Reply:
Surprising.  I didn't realize that was part of POSIX.  But FreeBSD, NetBSD, and OSX all allow fewer than six X's, so you should still change the docs.

Reply:
OpenBSD [docs](https://man.openbsd.org/mkstemp) says also says "at least six". The FreeBSD one (and OSX) is not very clear but still gives 6 Xs as an example. Are you sure this is not a minimum in all implementations? Moreover the docs for mkstemp is the same.

Reply:
Yep, there's definitely no requirement for a minimum number of Xs'.

Reply:
What should I put in the doc then?

Reply:
I would just say "a path whose rightmost characters contain some number of 'Xs'.  The 'Xs' will be replaced with a unique alphanumeric combination."  And in the next paragraph: "Err is returned is no temporary file could be created or if `path` had insufficient 'Xs'".